docs/support/unbricking/unbrick-ch341a.md

```diff
@@ -100,13 +100,16 @@ You don't need to do this if flashing a stock firmware backup created by the Fir
 
 1. For both the options below, we'll need to use the cbfstool (coreboot filesystem) binary, so let's download/extract that:
     * `wget https://mrchromebox.tech/files/util/cbfstool.tar.gz && tar -zxf cbfstool.tar.gz`
-    * Option 1: Extract VPD from the firmware on device
+    * Option 1: Extract VPD and HWID from the firmware on device
       * `sudo flashrom -p ch341a_spi -r badflash.rom`
       * `./cbfstool badflash.rom read -r RO_VPD -f vpd.bin`
-    * Option 2: Extract VPD from stock firmware backup created by Firmware Utility Script (this assumes the file has been copied into working directory)
+      * `./cbfstool backup.rom extract -n hwid -f hwid.txt`
+    * Option 2: Extract VPD and HWID from stock firmware backup created by Firmware Utility Script (this assumes the file has been copied into working directory)
       * `./cbfstool stock-firmware-<devicename>-<date>.rom read -r RO_VPD -f vpd.bin`
-      * Then we inject the VPD into the firmware image to be flashed.
+      * `./cbfstool stock-firmware-<devicename>-<date>.rom extract -n hwid -f hwid.txt`
+      * Then we inject the VPD and HWID into the firmware image to be flashed.
       * `./cbfstool <Shellball ROM/UEFI Full ROM filename> write -r RO_VPD -f vpd.bin`
+      * `./cbfstool <Shellball ROM/UEFI Full ROM filename> add -n hwid -f hwid.txt`
```

Comment:
This won't work, the HWID is not stored as a text file in CBFS in the stock firmware.
You need to use gbb_utility to write it to the GBB region of the stock firmware, just like my script does.

Reply:
I thought it uses cbfstool for writing since in the script uses cbfstool for injecting it into the coreboot file [at line 300 of firmware.sh](https://github.com/MrChromebox/scripts/blob/c795064bc51e100decdd38737764e2ae935e2ee6/firmware.sh#L300C5-L303C7)

Reply:
that's how I store it in the MrChromebox image, but only so it can be easily retrieved/restored to a shellball ROM if needed. If you look at https://github.com/MrChromebox/scripts/blob/c795064bc51e100decdd38737764e2ae935e2ee6/firmware.sh#L536 you can see how I extract it from the MrChromebox firmware and inject into a shellball. If restoring from a stock backup created by the script, that already has a valid HWID so no need to persist it in that case - only using a shellball/recovery image firmware.

Reply:
Ah ok, so for shellball it's gbb_utility but for uefi fullrom it's cbfstool?

Reply:
> Ah ok, so for shellball it's gbb_utility but for uefi fullrom it's cbfstool?

correct